### PR TITLE
feat: tl scraping function

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**  
This pull request introduces several enhancements and fixes to improve error handling, logging, and data extraction in the `functions-python` project. Key changes include robust error handling in the `process_dataset` function, updates to the `TransitFeedSyncProcessor` for better feed URL handling, and refactoring of test functions to align with the updated codebase.  

Closes https://github.com/MobilityData/product-tasks/issues/103)](https://github.com/MobilityData/product-tasks/issues/103

---

**What was fixed:**  
- Corrected handling of authentication information.  
- Enhanced generation of feed `stable_id` values.  
- Improved processing of GTFS real-time feeds, including proper management of entity types.  
- Fixed SQLAlchemy population for GTFS and GTFS-RT feeds (ensuring child and parent classes are properly populated).  
- Updated DataFrame processing to include all feeds (previously, some feeds were incorrectly filtered out).  
- Ensured correct message formatting when sending data to the batch processing function.  
- Updated triggers to account for `tld-*` feed IDs alongside `mdb-*` feed IDs.  

---

**Additional updates:**  
- Added infrastructure deployment code.  
- Performed end-to-end (e2e) testing for the first population of feeds.  

---

**Out of scope:**  
- E2E testing for subsequent runs: Since the TransitLand API does not provide updates between consecutive runs, this was not feasible to test at this stage. However, this should be prioritized before the next run to ensure consistent behavior.  

---

**Expected behavior:**  
- Successfully scrape over 1,200 feeds from the TransitLand API, including approximately 400 GTFS real-time feeds.  
image

--- 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by cka-y** [comment]
Added @AlfredNwolisa for visibility

File: functions-python/feed_sync_dispatcher_transitland/src/main.py [hunk 1]
```diff
@@ -345,12 +354,12 @@ def get_mbd_feed_url(
...
         :param db_session: SQLAlchemy session
         :param external_id: The external_id (feeds_onestop_id) from TransitLand
-        :param source: The source to filter by (e.g., 'TLD' for TransitLand)
+        :param source: The source to filter by (e.g., 'tld' for TransitLand)
         :return: feed_url in mbd if exists, otherwise None
         """
         results = (
-            db_session.query(Gtfsfeed)
-            .filter(Gtfsfeed.externalids.any(associated_id=external_id))
+            db_session.query(Feed)
```
Comment (davidgamez) [inline functions-python/feed_sync_dispatcher_transitland/src/main.py:361]: 👍

File: functions-python/feed_sync_process_transitland/src/main.py [hunk 2]
```diff
@@ -73,404 +37,156 @@ def log_message(level, message):
...
-        return None, None
-
-    def publish_to_batch_topic(self, payload: FeedPayload) -> None:
-        """
-        Publish feed to dataset batch topic
 
-        Args:
-            payload (FeedPayload): The feed payload to publish
-        """
+    def _update_feed(self, payload: FeedPayload, old_feed_id: Optional[str]) -> Feed:
```
Comment (davidgamez) [inline functions-python/feed_sync_process_transitland/src/main.py:120]: Rename this function to clarify that it is `deprecating` a feed in favor of a new feed.

**Review by davidgamez — APPROVED** [review]
🥇

**Pull request merged**